rfc-170-mobile-app-backend-stack.md

```diff
@@ -0,0 +1,80 @@
+---
+status: proposed
+implementation: proposed
+status_last_reviewed:
+---
+
+# Back end stack for GOV.UK App backends
+
+## Summary
+
+This RFC proposes a back end architecture and tech stack for internet-facing services (i.e. APIs) which underpin the GOV.UK mobile app. It draws on the existing work carried out on the Digital Identity programme to define an extensible, scalable architecture which meets the needs of a high-volume mobile application, and create a solid base for future development.
+
+The proposal is to adopt an AWS serverless stack, in line with DI, with the predominant implementation approach being to run Node.js Lambda functions to execute business logic.
+
+## Problem
+
+The new GOV.UK App is going to require back end services to enable many of its features. The app is a shared product with Digital Identity, where the existing ID Check app that they have in production will be superseded by a new, single app that incorporates both Digital Identity and GOV.UK features.
+
+The DI Mobile team have an existing set of back ends which are implemented in line with the Digital Identity architecture - namely, AWS serverless using Lambda functions (primarily Node.js with some Java). The decision making process behind this is recorded in the Digital Identity architecture repostitory (for example, [ADR-0001](https://github.com/govuk-one-login/architecture/blob/main/adr/0001-auth-hosting.md)) _(N.B. this is currently a private repo)_.
+
+As GOV.UK, we need to determine a tech stack for back end services which underpin the GOV.UK elements of the app.
+
+The following are key considerations we need to take into account:
+
+1. There is a single app, and it would be most straightforward to integrate the app with a single back end. This means we only need a single set of configuration rather than multiple settings for different features. Additionally, we can centralise concerns like logging, monitoring and alerting.
```

Comment:
I'm interested why exactly we'd need a single backend. Have we considered the implications DI running a separate API endpoint for the identity functionality, vs separate APIs for (potentially hosted in GOV.UK's platform) for other features. 

Clearly defined APIs for these supporting services could facilitate the decoupling of these backend services from the App team. I'm concerned that without this separation, the scope of the App team's responsibilities could expand excessively, encompassing a wide range of backend APIs. These could instead be developed and maintained independently by other relevant teams.

Reply:
Particularly if these backend services have the potential to be leveraged by not just the App.

Reply:
> Particularly if these backend services have the potential to be leveraged by not just the App.

Totally agree, I'm sure both AI and App have some common requirements for such an API.